cross-media-measurement/src/test/cc/wfa/measurement/common/crypto/liquid_legions_v2_encryption_utility_test.cc

```diff
@@ -1094,5 +1054,78 @@ TEST(FrequencyNoise, NonDeterministicNoiseShouldRandomizeTheResult) {
+  TestData test_data;
+  Sketch plain_sketch = CreateEmptyLiquidLegionsSketch();
+  // register 1: count = 6
+  // register 2: count = 3
+  // register 3: count = 1+2+3 = 6
+  // register 4: count = 3+4+5 = 12 (capped as kMaxFrequency+1, i.e. 11
+  // register 5: locally destroyed (key=-1), ignored
+  // register 6: destroyed by conflicting keys (601 and 602), ignored
+  // register 7: locally destroyed (key=-1) + normal register, ignored
+  //
+  // final result should be { 3->1/4, 6->2/4, 11->1/4 }
+  AddRegister(&plain_sketch, /*index=*/1, /*key=*/111, /*count=*/6);
+  AddRegister(&plain_sketch, /*index=*/2, /*key=*/222, /*count=*/3);
+  AddRegister(&plain_sketch, /*index=*/3, /*key=*/333, /*count=*/1);
+  AddRegister(&plain_sketch, /*index=*/3, /*key=*/333, /*count=*/2);
+  AddRegister(&plain_sketch, /*index=*/3, /*key=*/333, /*count=*/3);
+  AddRegister(&plain_sketch, /*index=*/4, /*key=*/444, /*count=*/3);
+  AddRegister(&plain_sketch, /*index=*/4, /*key=*/444, /*count=*/4);
+  AddRegister(&plain_sketch, /*index=*/4, /*key=*/444, /*count=*/5);
+  AddRegister(&plain_sketch, /*index=*/5, /*key=*/-1, /*count=*/2);
+  AddRegister(&plain_sketch, /*index=*/6, /*key=*/601, /*count=*/2);
+  AddRegister(&plain_sketch, /*index=*/6, /*key=*/602, /*count=*/2);
+  AddRegister(&plain_sketch, /*index=*/7, /*key=*/-1, /*count=*/2);
+  AddRegister(&plain_sketch, /*index=*/7, /*key=*/777, /*count=*/2);
+
+  std::string encrypted_sketch =
+      test_data.EncryptWithFlaggedKey(plain_sketch).value();
+
+  RegisterNoiseGenerationParameters reach_noise_parameters;
+  reach_noise_parameters.set_curve_id(kTestCurveId);
+  reach_noise_parameters.set_total_sketches_count(3);
+  reach_noise_parameters.set_contributors_count(kNumOfWorkers);
+  // resulted p = 0.716531, offset = 15.
+  // Random blind histogram noise.
+  *reach_noise_parameters.mutable_dp_params()->mutable_blind_histogram() =
+      NewDifferentialPrivacyParams(1, 1);
+  // resulted p = 0.716531, offset = 10.
+  // Random noise for publisher noise.
+  *reach_noise_parameters.mutable_dp_params()
+       ->mutable_noise_for_publisher_noise() =
+      NewDifferentialPrivacyParams(1, 1);
+  // resulted p ~= 0 , offset = 3.
+  // Deterministic reach dp noise.
+  *reach_noise_parameters.mutable_dp_params()->mutable_global_reach_dp_noise() =
+      NewDifferentialPrivacyParams(40, std::exp(-80));
+  *reach_noise_parameters.mutable_composite_el_gamal_public_key() =
+      test_data.client_el_gamal_public_key_;
+
+  FlagCountTupleNoiseGenerationParameters frequency_noise_params;
+  frequency_noise_params.set_maximum_frequency(kMaxFrequency);
+  frequency_noise_params.set_contributors_count(kNumOfWorkers);
+  // resulted p ~= 0, offset = 5, such that the number of frequency DP noise is
+  // almost a constant, and thus the result is deterministic.
+  *frequency_noise_params.mutable_dp_params() =
+      NewDifferentialPrivacyParams(40, std::exp(-80));
```

Comment:
"New" is a little misleading because it's associated with pointers/heap allocation. Consider renaming to "MakeDifferentialPrivacyParams".

Reply:
Done